gtests/CMakeLists.txt

```diff
@@ -46,3 +46,19 @@ gtest_discover_tests(test_nopython)
 add_custom_target(run_gtest
     COMMAND $<TARGET_FILE:test_nopython>
     DEPENDS test_nopython)
+
+# New binary target for testing take_along_axis_simd directly from C++
```

Comment:
Why do you need a new binary instead of using [`test_nopython` at L48](https://github.com/solvcon/modmesh/pull/641/files#diff-d7873e4ae849e77b02084c569b8d18ff4037158430e93f47869159c273a97b64R48) above?

Reply:
Cool